File: src/App/src/app/shared/models/conference-presentation-dto.ts
import { ConferenceSpeakerDto } from './conference-speaker-dto';

export interface ConferencePresentationDto {
    id: string;
    title: string;
    abstract: string;
    startDateTime: string;
    endDateTime: string;
    roomId: string;
    roomName: string;
    speakerIds: string[];
    speakers: ConferenceSpeakerDto[];
    externalId?: string;
}

export interface CreateConferencePresentationRequest {
    title: string;
    abstract: string;
    startDateTime: string;
    endDateTime: string;
    roomId: string;
    speakerIds: string[];
}

export interface UpdateConferencePresentationRequest {
    title: string;
    abstract: string;
    startDateTime: string;
    endDateTime: string;
    roomId: string;
    speakerIds: string[];
}

File: src/Conferences/HexMaster.Attendr.Conferences.Abstractions/Dtos/ConferencePresentationDto.cs
namespace HexMaster.Attendr.Conferences.Abstractions.Dtos;

public sealed record ConferencePresentationDto(
    Guid Id,
    string Title,
    string Abstract,
    DateTimeOffset StartDateTime,
    DateTimeOffset EndDateTime,
    Guid RoomId,
    string RoomName,
    IReadOnlyList<Guid> SpeakerIds,
    IReadOnlyList<ConferenceSpeakerDto> Speakers,
    string? ExternalId);
